### ISSUE TITLE
All-path proof with priority group results in an unclear error

### ISSUE THREAD

**Issue by yzhang90** [kframework/kore#1592]
I've tried the latest kore (c0937b13197802c5465f41ad4e2fc8a47ddb36b4) on the following definition and spec.
definition:
```
module TEST
  imports DOMAINS

  syntax Exp ::= Int
               | "Uop" "(" Int ")"
               | "Bop" "(" Int "," Int ")"

    configuration <T>
                    <k> $PGM:Exp </k>
                  </T>

  rule Uop(X) => 0  requires X >Int 10                     [priority(10)]
  rule Bop(X,Y) => 0 requires X >Int 10 andBool Y >Int 10  [priority(10)]

  // normal rules are of priority 50
  rule Uop(X) => 1  requires X >Int 0
  rule Bop(X, Y) => 1 requires X >Int 0 andBool Y >Int 0

  // owise rules are of priority 200
  rule Uop(X) => 2  [owise]
  rule Bop(X, Y) => 2 [owise]

endmodule
```
and spec:
```
module A-SPEC
  imports TEST

  rule <k> Uop(X) => 0 </k>
    requires X >Int 10

endmodule
```
kproves gives me the following error:
```
kore-exec: [2020-02-25 09:00:38.265896774] Error (ErrorException): Could not force pattern to sort 
'SortGeneratedTopCell{}'
, instead it has sort 
'SortGeneratedTopCell{}'
:
    /*  */ priorityLE50{}()
CallStack (from HasCallStack):
  error, called at src/Kore/Internal/TermLike.hs:603:6 in kore-0.0.1.0-5ymBeXBKB0u3tZRtNqnCe9:Kore.Internal.TermLike
  illSorted, called at src/Kore/Internal/TermLike.hs:673:26 in kore-0.0.1.0-5ymBeXBKB0u3tZRtNqnCe9:Kore.Internal.TermLike
  forceSortPredicate, called at src/Kore/Internal/TermLike.hs:597:14 in kore-0.0.1.0-5ymBeXBKB0u3tZRtNqnCe9:Kore.Internal.TermLike
  fullyOverrideSort, called at src/Kore/Internal/Predicate.hs:234:25 in kore-0.0.1.0-5ymBeXBKB0u3tZRtNqnCe9:Kore.Internal.Predicate
  coerceSort, called at src/Kore/Step/Rule/Simplify.hs:103:25 in kore-0.0.1.0-5ymBeXBKB0u3tZRtNqnCe9:Kore.Step.Rule.Simplify
  <T>
    <k>
      Uop ( X )
    </k>
  </T>
#And
  {
    X >Int 10
  #Equals
    true
  }

```

**Comment by traiansf** [comment]
@yzhang90 : I couldn't manage to reproduce, because `owise` for rules seem to not be encoded properly by the frontend, so an error is generated when parsing the rewrite rules.   

Nevertheless, if you set all priority attributes manually, it seems to work fine.
@ana-pantilie just submitted another PR which should fix the default priority, too. For owise, probably something needs to be done in the frontend.

**Comment by yzhang90** [comment]
@traiansf now the `owise` rule is encoded as priority 200 in the frontend. Is that the case on your machine? If not, maybe you need to update the k version.

**Comment by traiansf** [comment]
I was using the latest official front-end for KORE. Could you confirm though that the owise encoding to priority works for rewrite rules, too? Is the owise attribute removed when replaced?

**Comment by yzhang90** [comment]
the K frontend fkr KORE is not the latest one. (I checked and the correct encodings for owise rule is some commit after the commit that is used by KORE). On my machine, the owise rule is translated as follows:
```
// rule `<generatedTop>`(`<T>`(`<k>`(inj{Exp,KItem}(`Bop(_,_)_TEST_Exp_Int_Int`(X,Y))~>DotVar2)),DotVar0)=>`<generatedTop>`(`<T>`(`<k>`(inj{Int,KItem}(#token("2","Int"))~>DotVar2)),DotVar0) requires #token("true","Bool") ensures #token("true","Bool") [UNIQUE_ID(c70771ca4fde0ac68bea00d46a867cfe013af48793b1633be060ef658d10c98c) contentStartColumn(8) contentStartLine(21) org.kframework.attributes.Location(Location(21,8,21,22)) org.kframework.attributes.Source(Source(/home/zy/workspace/priority-test/./test.k)) org.kframework.definition.Production(syntax #RuleContent ::= #RuleBody [klabel(#ruleNoConditions) symbol()]) owise()]
  alias rule13LHS{}(SortGeneratedCounterCell{},SortK{},SortInt{},SortInt{}) : SortGeneratedTopCell{}
  where rule13LHS{}(VarDotVar0:SortGeneratedCounterCell{},VarDotVar2:SortK{},VarX:SortInt{},VarY:SortInt{}) :=
    \and{SortGeneratedTopCell{}} (
      \top{SortGeneratedTopCell{}}(), Lbl'-LT-'generatedTop'-GT-'{}(Lbl'-LT-'T'-GT-'{}(Lbl'-LT-'k'-GT-'{}(kseq{}(inj{SortExp{}, SortKItem{}}(LblBop'LParUndsCommUndsRParUnds'TEST'Unds'Exp'Unds'Int'Unds'Int{}(VarX:SortInt{},VarY:SortInt{})),VarDotVar2:SortK{}))),VarDotVar0:SortGeneratedCounterCell{})) []

  axiom{} \rewrites{SortGeneratedTopCell{}} (
    \and{SortGeneratedTopCell{}}(
      \not{SortGeneratedTopCell{}}(priorityLE50{}()),
      rule13LHS{}(VarDotVar0:SortGeneratedCounterCell{},VarDotVar2:SortK{},VarX:SortInt{},VarY:SortInt{})),
    \and{SortGeneratedTopCell{}} (
      \top{SortGeneratedTopCell{}}(), Lbl'-LT-'generatedTop'-GT-'{}(Lbl'-LT-'T'-GT-'{}(Lbl'-LT-'k'-GT-'{}(kseq{}(inj{SortInt{}, SortKItem{}}(\dv{SortInt{}}("2")),VarDotVar2:SortK{}))),VarDotVar0:SortGeneratedCounterCell{})))
  [org'Stop'kframework'Stop'attributes'Stop'Source{}("Source(/home/zy/workspace/priority-test/./test.k)"), org'Stop'kframework'Stop'definition'Stop'Production{}("syntax #RuleContent ::= #RuleBody [klabel(#ruleNoConditions) symbol()]"), contentStartLine{}("21"), contentStartColumn{}("8"), org'Stop'kframework'Stop'attributes'Stop'Location{}("Location(21,8,21,22)"), owise{}(), UNIQUE'Unds'ID{}("c70771ca4fde0ac68bea00d46a867cfe013af48793b1633be060ef658d10c98c")]
```
and owise attribute is not removed. but i do not add priority 200 either.

**Comment by ana-pantilie** [comment]
I removed the `owise` rules from the semantics to be sure that the `owise` encoding doesn't affect the result between the two frontend versions. So this is the modified semantics:

```
module TEST
  imports DOMAINS

  syntax Exp ::= Int
               | "Uop" "(" Int ")"
               | "Bop" "(" Int "," Int ")"

    configuration <T>
                    <k> $PGM:Exp </k>
                  </T>

  rule Uop(X) => 0  requires X >Int 10                     [priority(10)]
  rule Bop(X,Y) => 0 requires X >Int 10 andBool Y >Int 10  [priority(10)]

  // normal rules are of priority 50
  rule Uop(X) => 1  requires X >Int 0
  rule Bop(X, Y) => 1 requires X >Int 0 andBool Y >Int 0

endmodule
```

Running the spec with the `kore` frontend version doesn't throw this error and it results in `top`.
Running the spec with the latest frontend, commit `962d8e61985e4d8580685a9472857770a7e8e90d`, results in a weird sort coercion error, similar to the one you were seeing:
```
kore-exec: [2020-02-27 11:17:07.124617018] Error (ErrorException): Could not force pattern to sort
'SortGeneratedTopCell{}'
, instead it has sort
'SortGeneratedTopCell{}'
:
    /*  */ priorityLE10{}()
CallStack (from HasCallStack):
  error, called at src/Kore/Internal/TermLike.hs:604:6 in kore-0.0.1.0-5ymBeXBKB0u3tZRtNqnCe9:Kore.Internal.TermLike
  illSorted, called at src/Kore/Internal/TermLike.hs:674:26 in kore-0.0.1.0-5ymBeXBKB0u3tZRtNqnCe9:Kore.Internal.TermLike
  forceSortPredicate, called at src/Kore/Internal/TermLike.hs:598:14 in kore-0.0.1.0-5ymBeXBKB0u3tZRtNqnCe9:Kore.Internal.TermLike
  fullyOverrideSort, called at src/Kore/Internal/Predicate.hs:270:20 in kore-0.0.1.0-5ymBeXBKB0u3tZRtNqnCe9:Kore.Internal.Predicate
  binaryOperator, called at src/Kore/Internal/Predicate.hs:327:5 in kore-0.0.1.0-5ymBeXBKB0u3tZRtNqnCe9:Kore.Internal.Predicate
  makeAndPredicate, called at src/Kore/Step/Rule/Simplify.hs:104:27 in kore-0.0.1.0-5ymBeXBKB0u3tZRtNqnCe9:Kore.Step.Rule.Simplify
```
When comparing the two versions of `vdefinition.kore`, I observed that the `priorityLE10` term was missing in the `kore` frontend version. I don't know what this means, but what's stranger is that when running the same spec but with explicit priority (adding [priority(50)]) with the same (latest) frontend, the error doesn't appear anymore, even though the `priorityLE` rules seem identical (they're only missing the priority attribute in the default version, which is expected). I don't really know what to make of this, maybe I'm missing something but I probably need to dig deeper.

@traiansf , @ttuegel , any chance you might have any idea about what's going on here?

**Comment by ttuegel** [comment]
Fixed by #1622
